pkg/comp-functions/runtime/function_mgr.go

```diff
@@ -560,6 +560,18 @@ func (s *ServiceRuntime) SetDesiredKubeObserveObject(obj client.Object, objectNa
 	return s.SetDesiredComposedResourceWithName(kobj, objectName)
 }
 
+// SetDesiredKubeObserveObjectWithLabels takes any `runtime.Object`, puts it into a provider-kubernetes Object and then
+// adds it to the desired composed resources.
+func (s *ServiceRuntime) SetDesiredKubeObserveObjectWithLabels(obj client.Object, objectName string, labels map[string]string, refs ...xkube.Reference) error {
+
+	kobj, err := s.putIntoObjectWithLabels(true, obj, objectName, objectName, labels, refs...)
+	if err != nil {
+		return err
+	}
+
+	return s.SetDesiredComposedResourceWithName(kobj, objectName)
+}
+
```

Comment:
I see here good pattern for builder.

SetDesiredKubeObject().WithName().WithLabels().WithReferences()

does anyone else agree? Should we open a ticket to refactor that piece of code?

Reply:
I'd rather go with functional options like we already have with `SetDesiredKubeObject`. IMHO makes it easier to use than having a builder.